open-company/open-company-web
# Fix onboard drop off

Card: https://trello.com/c/p2grDSbd

Add a `status` key to the auth settings response when the user is logged in that contains an array of the following elements: `:password-required`, `:name-required`. It's used in conjunction with the storage entry point load to understand if the current user needs to be redirected to one of onboard/invitee flow because he dropped off.

The problem Stuart found was this: while accepting an invitation he abandoned the flow before filling in the first/last name so the post action was broken because the user had no name. Also another similar issue is this: close the window while onboarding at the profile setup step or the org setup step.
With this now you should be redirected to the onboard/invtee step that you left off.

This also should fix the issue of a user that abandon the signup/invitee flow before finishing setup his user or org.

Related orgs:
- https://github.com/open-company/open-company-storage/pull/122
- https://github.com/open-company/open-company-auth/pull/38

More accurate test steps are described here: https://beta.carrot.io/carrot/post/c5f5-4a57-83ba

To test:
- signup with email/password
- on the second step (user profile setup) open another window
- navigate to our homepage (localhost:3559 or staging.carrot.io)
- [x] are you redirected to the profile setup step? Good
- fill the user name and proceed
- on the third step (team setup) open another window
- navigate to our homepage (localhost:3559 or staging.carrot.io)
- [x] are you redirected to the team setup step? Good
- fill the team data and proceed
- invite another user from onboard
- accept the invitation in a new session
- before setting you password close the window
- open another window in the same session and go to the homepage
- [x] are you redirected to the password collect step for invitee flow? Good
- fill in your password
- before adding your name/avatar close the window and open another again at homepage
- [x] are you redirected to the user name collect step for invitee flow? Good
- now do forgot password for one the users to make sure it still works
- signup with slack
- invite someone from the slack org
- accept invitation
- [x] everything fine? Good
- add an email domain you can receive emails to
- signup with an email of that domain
- [x] do you get the email wall? Good
- go to your email and click the email verification link
- [x] could you get in? Everything fine? Good
